### PR TITLE
Discussion-rendering consumes `idApiUrl`

### PR DESCRIPTION
**Pull request by ioannakok**
Co-authored-by: Georges Lebreton <georges.lebreton@guardian.co.uk>

## Context
In order to support the identity's team work on migration we need to make sure they can use the CODE environment for testing. Currently, the flow below is broken:

1. A user who does not have a discussion username set
2. Tries to make a comment
3. They're taken to the 'set username' flow
4. When they hit submit, nothing happens on the page, and the POST call to `https://idapi.theguardian.com/user/me?api-key=dotcom-rendering` fails with a CORS error

image

image

5. The payload of that call is the setting of the new username, which means the call to set a new username doesn't work 

image

## What does this change?
Adds `idApiUrl` prop to discussion-rendering `App`. Its value will be coming from CAPI and passed to discussion-rendering [via DCR](https://github.com/guardian/dotcom-rendering/pull/6968).

## Why?
CAPI knows the correct url

| CODE | PROD |
|------|------|
| image | image |

## Link to supporting GH issue
#677

### CODE REVIEW

File: .changeset/strong-eyes-explode.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@guardian/discussion-rendering': patch
```
Comment (bryophyta) [inline .changeset/strong-eyes-explode.md:2]: if we're adding a new required property to the app, should this be a major change instead?

File: src/lib/api.tsx [hunk 2]
```diff
@@ -27,14 +28,17 @@ export const initialiseApi = ({
...
 }: {
 	baseUrl: string;
 	additionalHeaders: AdditionalHeadersType;
 	apiKey: string;
+	idApiUrl: string;
 }) => {
 	options.baseUrl = baseUrl || options.baseUrl;
 	options.headers = additionalHeaders || options.headers;
 	options.apiKey = apiKey || options.apiKey;
+	options.idApiUrl = idApiUrl || options.idApiUrl;
```
Comment (bryophyta) [inline src/lib/api.tsx:41]: is this fallback intended to guard against the case where the `idApiUrl` passed on line 31 is an empty string?

**Review by bryophyta — COMMENTED** [review]
this looks good to me overall! just a question about the semantic level of the change I think, and also I thought it would be good to clarify the intended use for the fallback value in `api.tsx`

File: .changeset/strong-eyes-explode.md (hunk 1 above)
Comment (ioannakok) [reply .changeset/strong-eyes-explode.md]: Yes, good point! I have updated the changeset.

File: src/lib/api.tsx (hunk 2 above)
Comment (ioannakok) [reply src/lib/api.tsx:41]: Good question! Yes, I believe that would be the empty string. I am not entirely clear about the full flow, i.e. how this value ends up in CAPI, but I'm guessing an empty string could happen, so we're adding this default case. Also, this was we are following the existing pattern.

File: .changeset/strong-eyes-explode.md (hunk 1 above)
Comment (bryophyta) [reply .changeset/strong-eyes-explode.md]: thanks!

**Review by bryophyta — APPROVED** [review]
looks good to me, thanks for clarifying the earlier questions!

**Pull request merged**